js/viewer.js

```diff
@@ -5,10 +5,20 @@ var videoViewer = {
 				return $.when();
 			} else {
 				this.videoJSLoaded = true;
-				var stylePath = OC.filePath('files_videoplayer', 'videojs', 'src/video-js.css');
+				var stylePath = OC.filePath('files_videoplayer', 'js', 'videojs/video-js.css');
 				$('head').append($('<link rel="stylesheet" type="text/css" href="' + stylePath + '"/>'));
-				var scriptPath = OC.filePath('files_videoplayer', 'videojs', 'src/video.js');
-				return $.getScript(scriptPath, function (xhr) {eval(xhr);});
+				var scriptPath = OC.filePath('files_videoplayer', 'js', 'videojs/video.js');
+
+				var deferred = $.Deferred();
```

Comment:
this is duplicating https://github.com/nextcloud/server/blob/b7767a51f140d3f588278f053f3a9e1eb84e4346/core/js/js.js#L364-L385, isn't it? Does `$.getScript` load the script differently?

Reply:
Yes. I have no idea why or how but it is different. With `$.getScript` I did not have a `videojs` funtion after loading. This way i do.

Reply:
@ChristophWurst but it probably would make sense to look into a generic approach here indeed. This is just the quick fix for 15.

Reply:
I see, they are indeed a bit different. Let's use this fix for 15 now and look into a proper npm setup for 16 where we can use dependabot and video.js from https://www.npmjs.com/package/video.js.

Reply:
We could integrate that with the slideshow/big picture we discussed in https://github.com/nextcloud/server/issues/12382

Reply:
Yeah, sounds good. So let's npm'ize this repo for 16, switch to Vue and use a standardized component for the overlay.